### PR TITLE
Fix link, test, and update binaries version

### LINKED ISSUE #649: "Bonmin" link in "Binary Packages" documentation points to Petsc website
On the following page:
https://idaes-pse.readthedocs.io/en/stable/getting_started/binaries.html
under "Main packages", the "bonmin" bullet. I assume this is not intentional.

### PR DESCRIPTION
**Pull request by eslickj**
## Fixes #649 

This fixes the Bonmin link, removes an extra test, and bumps the binary version to get a few minor PETSc fixes.

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: idaes/core/solvers/tests/test_petsc.py [hunk 1]
```diff
@@ -26,6 +26,15 @@
 from idaes.core.solvers.features import dae
 
 def rp_example():
+    """This example is done multiple ways to test a few common errors where the
+    integrator and fully time-discretized problem differ and alternative working
+    formulations.
+
+    The PETSc utilities raise an exception when a differential variable is fixed.
+    While this is okay for the fully time-discretized problem, for the integrator
+    it implies the derivative is 0.  While this problem can be solved with the
```
Comment (Robbybp) [inline idaes/core/solvers/tests/test_petsc.py:35]: I suggest changing to: "While this is okay for the fully time-discretized problem, the integrator will not correctly link a fixed differential variable (at non-initial time points) with a time derivative."

File: idaes/core/solvers/tests/test_petsc.py [hunk 2]
```diff
@@ -68,11 +86,22 @@ def x_eq_rule(m, t):
...
 
     return m
 
 def rp_example3():
+    """ This example is done multiple ways to test a few common errors where the
+    integrator and fully time-discretized problem differ, and alternative working
+    formulations.
+
+    Another way to formulate this problem is to fix the derivative. This doesn't
+    work for the integrator because it lose the association between the
```
Comment (Robbybp) [inline idaes/core/solvers/tests/test_petsc.py:100]: Typo: loses

**Review by Robbybp — CHANGES_REQUESTED** [review]
Minor wording change request, then this is good.

File: idaes/core/solvers/tests/test_petsc.py (hunk 1 above)
Comment (Robbybp) [reply idaes/core/solvers/tests/test_petsc.py:35]: The issue is that the derivative should not necessarily be zero. As long as this is corrected, I'm fine.

Comment (eslickj) [reply idaes/core/solvers/tests/test_petsc.py]: Sounds good.  Updated wording.

**Review by Robbybp — APPROVED** [review]
Looks good

**Review by blnicho — APPROVED** [review]

**Pull request merged**